public/static/app/common/WhitelistDataService.test.js

```diff
@@ -0,0 +1,247 @@
+describe('WhitelistDataServiceSpec', function () {
+  beforeEach(module('MCI'));
+
+  const project = 'sys-perf';
+  const variant = 'linux-standalone';
+  const task = 'bestbuy_agg';
+  const test_name = ' distinct_types_no_predicate-useAgg';
+  const db_perf = 'perf';
+  const coll_whitelist = 'whitelisted_outlier_tasks';
+  const query = {'project': project, 'variant': variant, 'task': task, 'test': test_name};
+  const mongo_filter = {'_id': 1};
+  let service;
+  let $db;
+  const createTaskRevision = (i) => {
+    i = i || 1;
+    return {'revision': 'revision ' + i, 'project': 'project ' + 1, 'variant': 'variant ' + i, 'task': 'task ' + i, 'random': i}
+  };
+  const createTaskRevisionQuery = (i) => {
+    i = i || 1;
+    return {'revision': 'revision ' + i, 'project': 'project ' + 1, 'variant': 'variant ' + i, 'task': 'task ' + i}
+  };
+  const setUpTaskRevisions = (count) => _.range( count || 1).map(createTaskRevision);
+  const setUpTaskRevisionQueries = (count) => _.range( count || 1).map(createTaskRevisionQuery);
+
+  /**
+   *  SetUp the base environment / closure. You can call further beforeEach blocks,
+   *  but the state will be 'set' on injection.
+   */
+  beforeEach(() => {
+    stitch = {};
+    $window = {project: project};
+    module($provide => {
+      Stitch = {
+        use: () => Stitch,
+        query: () => Stitch,
+        then: () => Stitch,
+      };
+      $provide.value('$window', $window);
+      $provide.value('Stitch', Stitch);
+    });
+  });
+
+  describe('addWhitelist', () => {
+    describe('query handling', () => {
+      let $q;
+      let deferred;
+      beforeEach(() => {
+        module($provide => {
+          $db = {
+            db: () => $db,
+            collection: () => $db,
+            execute: () => $db,
+            then: () => $db,
+            updateOne: () => $db,
+          };
+          spyOn($db, 'db').and.callThrough();
+          spyOn($db, 'collection').and.callThrough();
+          spyOn($db, 'updateOne').and.callThrough();
+          spyOn($db, 'execute').and.callThrough();
+
+          Stitch = {
+            use: () => Stitch,
+            query: (cb) => cb($db),
+            then: () => Stitch,
+          };
+          $provide.value('Stitch', Stitch);
+        });
+
+        inject($injector => {
+          $q = $injector.get('$q');
+          deferred = $q.defer();
+          spyOn($q, 'all').and.returnValue(deferred.promise);
+
+          service = $injector.get('WhitelistDataService')
+        });
+
+
+      });
+      it('should process a single task_revisions', function () {
+        const task_revisions = setUpTaskRevisions();
+        const query  = createTaskRevisionQuery();
+
+        service.addWhitelist(task_revisions);
+        expect($db.db).toHaveBeenCalledWith(db_perf);
+        expect($db.collection).toHaveBeenCalledWith(coll_whitelist);
+        expect($db.updateOne).toHaveBeenCalledWith(query, task_revisions[0], {upsert: true});
+      });
```

Comment:
[nit] In this file, it seems like sometimes we have a blank line between 'it' blocks and sometimes we don't. Can we pick one and do it everywhere? I'd vote for having a single blank line between 'it' blocks, but mostly care that we are consistent.

Reply:
Added blank lines between block.

But the real solution here is a linter.